impeller/renderer/backend/vulkan/binding_helpers_vk.cc

```diff
@@ -45,36 +51,35 @@ static bool BindImages(const Bindings& bindings,
     image_info.imageLayout = vk::ImageLayout::eShaderReadOnlyOptimal;
     image_info.sampler = sampler.GetSampler();
     image_info.imageView = texture_vk.GetImageView();
-    images.push_back(image_info);
+    image_workspace[image_offset++] = image_info;
 
     vk::WriteDescriptorSet write_set;
     write_set.dstSet = vk_desc_set;
     write_set.dstBinding = slot.binding;
     write_set.descriptorCount = 1u;
     write_set.descriptorType = vk::DescriptorType::eCombinedImageSampler;
-    write_set.pImageInfo = &images.back();
+    write_set.pImageInfo = &image_workspace[image_offset - 1];
 
-    writes.push_back(write_set);
+    write_workspace[write_offset++] = write_set;
```

Comment:
A somewhat hypothetical issue but there are no guards around writes past the end of the buffer. Debugging memory corruptions might get hairy. Perhaps `write_workspace[std::max(write_offset++, write_workspace.size() - 1u)] = <blah>;`. Or do a bounds check and propagate the error. Here an elsewhere.